app/domain/authentication/authn_k8s/kubectl_exec.rb

```diff
@@ -26,106 +27,68 @@ def channel_names
         @msg.type
       end
-      
+
       def data
         @msg.data[1..-1]
       end
 
       def channel_name
         self.class.channel_names[channel_number]
       end
-      
+
       def channel_number
         unless @msg.respond_to?(:data)
           return self.class.channel_number_from_name('error') 
         end
-        
-        @msg.data[0..0].bytes.first
-      end
-    end
-
-    # Utility class for storing messages received during websocket communication.
-    class MessageLog
-      attr_reader :messages
-      
-      def initialize
-        @messages = Hash.new { |hash,key| hash[key] = [] }
-      end
-
-      def save_message(wsmsg)
-        channel_name = wsmsg.channel_name
-        
-        unless channel_name
-          raise "Unexpected channel: #{wsmsg.channel_number}"
-        end
-        
-        @messages[channel_name.to_sym] << wsmsg.data
-      end
 
-      def save_error_string(str)
-        @messages[:error] << str
+        @msg.data[0..0].bytes.first
       end
     end
-    
-    class KubectlExec
-      # logger: an object responding to `debug`
-      # kubeclient: Kubeclient::Client from "kubeclient" gem
-      #
-      def initialize(
-        pod_name:,
-        pod_namespace:,
-        logger:,
-        kubeclient:,
-        container: 'authenticator',
-        timeout: 5.seconds
-      )
-        @pod_name = pod_name
-        @pod_namespace = pod_namespace
-        @container = container
-        @timeout = timeout
-        @logger = logger
-        @kubeclient = kubeclient
 
+    KubectlExec = CommandClass.new(
+      dependencies: { logger: Rails.logger,
+                      k8s_object_lookup: K8sObjectLookup,
+                      timeout: 5.seconds },
+      inputs: %i( pod_namespace
+                  pod_name
+                  container
+                  cmds
+                  body
+                  stdin )
+    ) do
+      def call
         @message_log = MessageLog.new
         @channel_closed = false
-      end
 
-      def execute(cmds, body: "", stdin: false)
-        url = server_url(cmds, stdin)
-        headers = @kubeclient.headers.clone
+        url = server_url(@cmds, @stdin)
+        headers = kubeclient.headers.clone
         ws_client = WebSocket::Client::Simple.connect(url, headers: headers)
 
-        add_websocket_event_handlers(ws_client, body, stdin)
+        add_websocket_event_handlers(ws_client, @body, @stdin)
 
         wait_for_close_message
 
         raise CommandTimedOut.new(@container, @pod_name) unless @channel_closed
-        
+
         # TODO: raise an `WebsocketServerFailure` here in the case of ws :error
 
         @message_log.messages
       end
-      
-      def copy(path, content, mode)
-        execute(
-          [ 'tar', 'xvf', '-', '-C', '/' ],
-          stdin: true,
-          body: tar_file_as_string(path, content, mode)
-        )
-      end
-      
+
+      # on_* methods are public since ws_client needs public callbacks that it will
```

Comment:
I think you could remove the `that it will invoke` from this comment to fit it on one line. Why else would it need public callbacks unless it was planning to invoke them. :)